src/main/java/com/google/api/gax/grpc/OperationFuture.java

```diff
@@ -0,0 +1,372 @@
+   * Operation.error if not successful.
+   */
+  @Override
+  public final ResponseT get() throws InterruptedException, ExecutionException {
+    return finalResultFuture.get();
+  }
+
+  /**
+   * Waits on the polling loop on Operations.GetOperation, and once Operation.done
+   * is true, then returns Operation.response if successful or throws
+   * ExecutionException containing an ApiException with the status code set to
+   * Operation.error if not successful.
+   */
+  @Override
+  public final ResponseT get(long timeout, TimeUnit unit)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    return finalResultFuture.get(timeout, unit);
+  }
+
+  /**
+   * Returns true if the operation has been cancelled.
+   */
+  @Override
+  public final boolean isCancelled() {
+    return finalResultFuture.isCancelled();
+  }
+
+  /**
+   * Issues Operations.GetOperation and returns value of Operation.done.
+   * @see java.util.concurrent.Future#isDone()
+   */
+  @Override
+  public final boolean isDone() {
+    return finalResultFuture.isDone();
+  }
+
+  /**
+   * Enters a polling loop on Operations.GetOperation, and once Operation.done
+   * is true, notifies the listener.
+   * Normally, Futures.addCallback would be used instead of calling this directly.
+   */
+  @Override
+  public final void addListener(Runnable listener, Executor executor) {
+    finalResultFuture.addListener(listener, executor);
+  }
+
+  /**
+   * Returns the value of Operation.name from the initial Operation object returned
+   * from the initial call to start the Operation.
+   * Blocks if the initial call to start the Operation hasn't returned yet.
+   */
+  public final String getOperationName() throws InterruptedException, ExecutionException {
+    return startOperationFuture.get().getName();
+  }
+
+  /**
+   * Returns the value of Operation.metadata from the initial Operation object
+   * returned from the initial call to start the Operation.
+   * Blocks if the initial call to start the Operation hasn't returned yet.
+   */
+  public final Any getMetadata() throws InterruptedException, ExecutionException {
+    return startOperationFuture.get().getMetadata();
+  }
+
+  /**
+   * Returns the data from the last call to OperationsApi.GetOperation (or if only
+   * the initial API call has been made, the data from that call).
+   * Blocks if the initial call to start the Operation hasn't returned yet.
+   */
+  public final Operation getLastOperationData() throws InterruptedException, ExecutionException {
+    return startOperationFuture.get();
+  }
+
+  /**
+   * Awaits any ongoing asynchronous work.
+   */
+  public final void awaitAsyncCompletion() throws InterruptedException {
+    asyncCompletionLatch.await();
+  }
+
+  /**
+   * Awaits any ongoing asynchronous work.
+   */
+  public final void awaitAsyncCompletion(long timeout, TimeUnit unit) throws InterruptedException {
```

Comment:
We have been using Duration in other places before. Shall we keep it consistent?